solver/src/http_solver.rs

```diff
@@ -127,39 +120,50 @@ impl HttpSolver {
     }
 
     // Maps string based amm index from solver api
-    fn amms<'a>(&self, orders: &'a [Liquidity]) -> HashMap<String, &'a AmmOrder> {
+    fn amms(&self, orders: Vec<AmmOrder>) -> HashMap<String, AmmOrder> {
         orders
-            .iter()
-            .filter_map(|liquidity| match liquidity {
-                Liquidity::Limit(_) => None,
-                Liquidity::Amm(amm) => Some(amm),
-            })
+            .into_iter()
             .enumerate()
             .map(|(index, amm)| (index.to_string(), amm))
             .collect()
     }
 
     // Maps string based amm index from solver api
-    fn amm_models<'a>(
-        &self,
-        amms: &HashMap<String, &'a AmmOrder>,
-    ) -> HashMap<String, UniswapModel> {
+    fn amm_models(&self, amms: &HashMap<String, AmmOrder>) -> HashMap<String, UniswapModel> {
         amms.iter()
             .map(|(index, amm)| {
                 let uniswap = UniswapModel {
                     token1: self.token_to_string(&amm.tokens.get().0),
                     token2: self.token_to_string(&amm.tokens.get().1),
                     balance1: amm.reserves.0,
                     balance2: amm.reserves.1,
-                    // TODO use AMM fee
-                    fee: 0.003,
+                    fee: *amm.fee.numer() as f64 / *amm.fee.denom() as f64,
                     mandatory: false,
                 };
                 (index.clone(), uniswap)
             })
             .collect()
     }
 
+    fn prepare_model(&self, liquidity: Vec<Liquidity>) -> PreparedModel {
+        let tokens = self.tokens(liquidity.as_slice());
+        let orders = split_liquidity(liquidity);
+        let limit_orders = self.orders(orders.0);
+        let amm_orders = self.amms(orders.1);
+        let model = BatchAuctionModel {
+            tokens: self.token_models(&tokens),
+            orders: self.order_models(&limit_orders),
```

Comment:
It's a bit confusing to me to have `self.orders`, `self.order_models`, etc. (their separation is not really clear). 

Could we make the function names more explicit (e.g. order_list_to_map) and do they even have to be on self? Maybe we can even use the Into trait to convert `HashMap<String, AmmOrder>` into `HashMap<String, UniswapModel>`?

Reply:
I've changed the function names from `self.orders` to `self.map_orders_for_solver` and added another explanatory comment to `prepare_model` where these functions are used. I feel that is to make sense of this.
They don't have to be on self now but this will change when we want to query (and cache) the token decimals and name. When it is clear how that works out with self we can move what still doesn't need self.